pinot-broker/src/main/java/org/apache/pinot/broker/api/AccessControl.java

```diff
@@ -47,4 +47,13 @@ default boolean hasAccess(RequesterIdentity requesterIdentity) {
    * @return {@code true} if authorized, {@code false} otherwise
    */
   boolean hasAccess(RequesterIdentity requesterIdentity, BrokerRequest brokerRequest);
+
+  /**
+   * Determine whether authentication is required for annotated (controller) endpoints only
```

Comment:
Could you elaborate on this? Why do we need to mention controller endpoints in the pinot-broker module?

Reply:
Broker endpoints like `client/sql` (the one taking SQL queries) are doing access controls inside the endpoint. The endpoints takes one AccessControlFactory, and then calls AccessControl interface actively to verify.

There is no need for those endpoints to be verified twice since they have access control baked in it already.